### PR TITLE
Validation of settings is done at node agent service.

### PR DESCRIPTION
**Pull request by khandelwalbrijesh**

### CODE REVIEW

File: src/PatchOrchestrationApplication/NodeAgentService/src/NodeAgentService.cs [hunk 1]
```diff
@@ -80,11 +83,22 @@ internal void OnConfigurationPackageModified(object sender, PackageModifiedEvent
...
+            try
             {
-                this.ModifySettings(package.Settings.Sections[settingsSectionName]);
+                NtServiceConfigurationUtility.CreateConfigurationForNtService(package, settingsDestinationPath, this.fabricClient, this.Context);
+                if (package.Settings != null && package.Settings.Sections.Contains(settingsSectionName))
+                {
+                    this.ModifySettings(package.Settings.Sections[settingsSectionName]);
+                }
+                string healthMessage = "Settings validation was successful.";
+                HealthManagerHelper.PostServiceHealthReport(this.fabricClient, this.Context, SettingsValidationProperty, healthMessage, HealthState.Ok);
```
Comment (raunakpandya) [inline src/PatchOrchestrationApplication/NodeAgentService/src/NodeAgentService.cs]: >PostServiceHealthReport [](start = 36, length = 23)

This one you dont need to post with -1

File: src/PatchOrchestrationApplication/NodeAgentService/src/NTServiceConfigurationUtility.cs [hunk 2]
```diff
@@ -35,6 +38,76 @@ internal static void CreateConfigurationForNtService(ConfigurationPackage packag
...
+                paramName = "AcceptWindowsUpdateEula";
+                if (configurationSection.Parameters.Contains(paramName))
+                {
+                    ValidateParameter<bool>(paramName, configurationSection.Parameters[paramName].Value, fabricClient, serviceContext);
+                }
+            }
+        } 
+
+
+        private static void ValidateParameter<T> (string paramName, string value, FabricClient fabricClient, ServiceContext serviceContext)
```
Comment (raunakpandya) [inline src/PatchOrchestrationApplication/NodeAgentService/src/NTServiceConfigurationUtility.cs]: >ValidateParameter [](start = 28, length = 17)

Where are we validating the WU frequency? That is the most common setting people modify and add incorrectly. Either specify days as 29 or incorrect format.

Comment (khandelwalbrijesh) [reply src/PatchOrchestrationApplication/NodeAgentService/src/NTServiceConfigurationUtility.cs]: It is validated at NodeAgentNTService. It was a big code, so it did not make much sense to replicate it at NodeAgentService. But if someone inputs incorrect value for WUFrequency, we will post warning on the service.

Comment (raunakpandya) [reply src/PatchOrchestrationApplication/NodeAgentService/src/NTServiceConfigurationUtility.cs]: Where are we posting that warning?

---
In reply to: [282805214](https://github.com/microsoft/Service-Fabric-POA/pull/16#discussion_r282805214) [](ancestors = 282805214)

**Pull request merged**